tests/next_tests/integration_tests/feature_tests/ffront_tests/test_compiled_program.py

```diff
@@ -436,6 +436,62 @@ def test_compile_variants_not_compiled(cartesian_case, compile_variants_testee):
         )
 
 
+def test_compile_variants_not_compiled_but_jit_enabled_on_call(
+    cartesian_case, compile_variants_testee
+):
+    # disable jit on the program
+    object.__setattr__(compile_variants_testee, "enable_jit", False)
+
+    field_a = cases.allocate(cartesian_case, compile_variants_testee, "field_a")()
+    field_b = cases.allocate(cartesian_case, compile_variants_testee, "field_b")()
+    out = cases.allocate(cartesian_case, compile_variants_testee, "out")()
+
+    compile_variants_testee(
```

Comment:
Shouldn't be also tested here that if `enable_hit` is  not passed  at runtime the call fails?